springwolf-plugins/springwolf-amqp-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/RabbitChannelsScanner.java

```diff
@@ -13,15 +16,27 @@
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
 import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Slf4j
 @Service
 public class RabbitChannelsScanner extends AbstractChannelScanner<RabbitListener>
         implements ChannelsScanner, EmbeddedValueResolverAware {
 
+    private final Map<String, Binding> bindingsMap;
     private StringValueResolver resolver;
 
+    public RabbitChannelsScanner(List<Binding> bindings) {
+        bindingsMap = bindings.stream()
+                .filter(Binding::isDestinationQueue)
+                .collect(Collectors.toMap(Binding::getDestination, Function.identity()));
```

Comment:
Just taking all the bindings from the context and extracting the information about the exchange/routing key from it.
This will not work only in the case when the queue binding is declared outside of the application (on the RabbitMQ directly, or in some other application), and it's not subscribed to outside of the application - which would be highly unusual I'd say.